### ISSUE TITLE
Crash when loading glb model files via addStyleModel

### ISSUE THREAD

**Issue by MichaelDanielTom** [mapbox/mapbox-maps-ios#2168]
## Environment

- Xcode version: 15.3
- iOS version: 17.4
- Devices affected: iPhone
- Maps SDK Version: 11.3.0

## Observed behavior and steps to reproduce

Crash when calling [`map.addStyleModel(modelId:modelUri:)`](https://docs.mapbox.com/ios/maps/api/11.3.0/documentation/mapboxmaps/stylemanager/addstylemodel(modelid:modeluri:)) when the loaded file from the modelUri is a glb. Loading glbs via this method worked fine up to 11.2.0 and just started crashing in 11.3.0-beta.1. The crash appears to be in `MapboxCoreMaps` with an `EXC_BAD_ACCESS`.  Here are two parts of the stack where the error occurred:

```#0	0x000000010fb16f8d in mbgl::models::cgltf::GltfDataDeleter::operator()(cgltf_data*) const ()```
```
MapboxCoreMaps`std::__1::__function::__func<std::__1::function<void ()> mapbox::common::bindTaskAndReply<mbgl::models::ModelLoader::processModelIfAllSidecarsLoaded(std::__1::shared_ptr<mbgl::models::ModelRawData> const&, mbgl::models::ModelObserver&)::$_1, mbgl::models::ModelLoader::processModelIfAllSidecarsLoaded(std::__1::shared_ptr<mbgl::models::ModelRawData> const&, mbgl::models::ModelObserver&)::$_2>(mbgl::models::ModelLoader::processModelIfAllSidecarsLoaded(std::__1::shared_ptr<mbgl::models::ModelRawData> const&, mbgl::models::ModelObserver&)::$_1 const&, mbgl::models::ModelLoader::processModelIfAllSidecarsLoaded(std::__1::shared_ptr<mbgl::models::ModelRawData> const&, mbgl::models::ModelObserver&)::$_2 const&)::'lambda'(), std::__1::allocator<std::__1::function<void ()> mapbox::common::bindTaskAndReply<mbgl::models::ModelLoader::processModelIfAllSidecarsLoaded(std::__1::shared_ptr<mbgl::models::ModelRawData> const&, mbgl::models::ModelObserver&)::$_1, mbgl::models::ModelLoader::processModelIfAllSidecarsLoaded(std::__1::shared_ptr<mbgl::models::ModelRawData> const&, mbgl::models::ModelObserver&)::$_2>(mbgl::models::ModelLoader::processModelIfAllSidecarsLoaded(std::__1::shared_ptr<mbgl::models::ModelRawData> const&, mbgl::models::ModelObserver&)::$_1 const&, mbgl::models::ModelLoader::processModelIfAllSidecarsLoaded(std::__1::shared_ptr<mbgl::models::ModelRawData> const&, mbgl::models::ModelObserver&)::$_2 const&)::'lambda'()>, void ()>::operator():
```

## Expected behavior

glb model files should load similar to gltf files.

## Notes / preliminary analysis

It looks like the was a change in mapbox-maps-ios 11.3.0 to use [cgltf](https://github.com/jkuhlmann/cgltf) instead of tinygltf, so perhaps that's related? It looks like both libraries have support for glb and gltf format.

**Comment by jtorresfabra** [comment]
Hi @MichaelDanielTom is this happening with a model in particular that you can share? Or is it happening for you with every model?

**Comment by MichaelDanielTom** [comment]
@jtorresfabra Upon further inspection, it looks like most of the glb models are working as expected (my bad for first assuming it was a problem with all glbs), and the crash is happening with [this model](https://cdn2.radiatetheworldcf.com/media/MapModel_images/ae195d9d-c53a-4205-883c-01c75a77089c.glb) in particular. Not entirely sure what's different between this model and the others that are working but hopefully it's helpful!

**Comment by MichaelDanielTom** [comment]
@jtorresfabra It looks like with that model, if I export that model to glb from Blender with the `Compression` setting checked, mapbox will crash, but exporting the same model without compression checked will work ok. Here are the settings in the Blender gltf export extension that will crash mapbox (unchecking compression makes it work with mapbox).
![image](https://github.com/mapbox/mapbox-maps-ios/assets/5520838/aec8e451-9d3f-45b2-a608-4fc91449b3f8)

**Comment by jtorresfabra** [comment]
Ok, just detected the problem. Your model shares indices between meshes Cube.10 and Cube.11 and when releasing them in the draco compression path it was causing a double free. Fix implemented, and will reach the next release hopefully. For the time being, either you can avoid to use draco compression or use different indices array for each primitive. 

Thanks a lot for the report and the data!

I will close this issue once the fix is released.

**Comment by MichaelDanielTom** [comment]
Thank you so much @jtorresfabra!

**Comment by jtorresfabra** [comment]
The fix was included in the stable 11.4 release, closing this ticket. Thanks a lot!
